File: GRAPH/C++/NON WEIGHTED/Unirected_Graph_Adjacency_Matrix.cpp
#include <bits/stdc++.h>
using namespace std;

void add_edge(vector<vector<int>> &mat, int i, int j){
    if (mat[i][j] == 0 ){
        mat[i][j] = 1;
        mat[j][i] = 1;
    }
};

void display_matrix(vector<vector<int>> &mat){
    int v = mat.size();
    for(int i=0; i<v; i++){
        for(int j=0; j<v; j++)
            cout<<mat[i][j]<<' ';
        cout<<endl;
    }
};

int main(){
    int v = 4;
    vector<vector<int>> mat(v, vector<int>(v, 0));
    
    add_edge(mat, 0, 1);
    add_edge(mat, 0, 2);
    add_edge(mat, 1, 2);
    add_edge(mat, 2, 3);

    display_matrix(mat);
    return 0;
}

File: GRAPH/PYTHON/GRAPHS USING OOPS/Unirected_Graph_Adjacency_Matrix.py
class Graph: 
    def __init__(self):
        v = 3
        self.matrix = [[0]*v for _ in range(v)]
    
    def add_edge(self, i, j):
        self.matrix[i][j] = 1
        self.matrix[j][i] = 1

    def display_matrix(self):
        for row in self.matrix:
            print(' '.join(map(str, row)))

g = Graph()
g.add_edge(0, 1)
g.add_edge(0, 2)
g.add_edge(1, 0)
g.add_edge(1, 1)
g.add_edge(2, 1)

g.display_matrix()